agent/unit_state.go

```diff
@@ -71,12 +73,19 @@ type UnitStatePublisher struct {
 // them to the Registry every 5s. Heartbeat objects are also published as they
 // are received on the channel.
 func (p *UnitStatePublisher) Run(beatchan <-chan *unit.UnitStateHeartbeat, stop chan bool) {
+	var publishPeriod time.Duration
+	if p.ttl > 30*time.Second {
```

Comment:
to give it enough margin timewise to publish the state before the ttl kills it 
example:
for ttl => 15 second, publishPeriod => 7.5 secs
for: ttl => 10min, publishedPeriod => 9min 50 sec